pkg/rear-controller/gateway/utils.go

```diff
@@ -35,7 +35,15 @@ func selectorToQueryParams(selector models.Selector) (string, error) {
 	case models.K8SliceNameDefault:
 		k8sliceSelector := selector.(models.K8SliceSelector)
 		return encodeK8SliceSelector(k8sliceSelector)
-	// TODO: Implement the other selector types
+	case models.VMNameDefault:
+		// TODO (VM): Implement the VM selector type
+		return "", fmt.Errorf("unsupported selector type %s", selector.GetSelectorType())
+	case models.ServiceNameDefault:
+		serviceSelector := selector.(models.ServiceSelector)
+		return encodeServiceSelector(serviceSelector)
+	case models.SensorNameDefault:
+		// TODO (Sensor): Implement the Sensor selector type
+		return "", fmt.Errorf("unsupported selector type %s", selector.GetSelectorType())
```

Comment:
Same as before, shouldn't this be the default?

Reply:
This code is basically a placeholder for the future implementation, an easier way to implement what is missing while keeping the code reasonably tidy.